src/app/core/form-dialog/form-dialog.service.ts

```diff
@@ -61,7 +60,9 @@ export class FormDialogService {
     const dialogWrapper = dialogRef.componentInstance.formDialogWrapper;
     dialogWrapper.readonly = this.ability.cannot("update", entity);
 
-    dialogWrapper.close.subscribe((res) => dialogRef.close(res));
+    dialogWrapper.close
+      .pipe(untilDestroyed(dialogWrapper))
```

Comment:
does this work without some `@UntilDestroy()` annotation at a component? Or do the passed components all have that? (probably difficult to ensure via typing?)

Reply:
The passed component is the `FormDialogWrapperComponent` which has the annotation. The tests needed to be adjusted because they use a mock component.